common/SmokeTests/SmokeTest/Update-Dependencies.ps1

```diff
@@ -31,63 +37,93 @@ function Log-Warning($message) {
     }
 }
 
-Register-PackageSource `
-    -Name $NIGHTLY_FEED_NAME `
-    -Location $NIGHTLY_FEED_URL `
-    -ProviderName Nuget `
-    -ErrorAction SilentlyContinue `
-
-# List all packages from the source specified by $FeedName. Packages are sorted
-# ascending by version according to semver rules (e.g. 4.0.0-preview.1 comes
-# before 4.0.0) not lexicographically.
-# Packages cannot be filtered at this stage because the sleet feed to which they
-# are published does not support filtering by name.
-$allPackages = Find-Package -Source $NIGHTLY_FEED_NAME -AllVersion -AllowPrereleaseVersions
+function GetAllNightlyPackages {
+    Register-PackageSource `
+        -Name $NIGHTLY_FEED_NAME `
+        -Location $NIGHTLY_FEED_URL `
+        -ProviderName Nuget `
+        -ErrorAction SilentlyContinue `
+
+    # List all packages from the source specified by $FeedName. Packages are sorted
+    # ascending by version according to semver rules (e.g. 4.0.0-preview.1 comes
+    # before 4.0.0) not lexicographically.
+    # Packages cannot be filtered at this stage because the sleet feed to which they
+    # are published does not support filtering by name.
+    return Find-Package -Source $NIGHTLY_FEED_NAME -AllVersion -AllowPrereleaseVersions
+}
 
-$baselineVersionDate = $null;
+function GetAllPackages {
+    if ($Nightly) {
+        return GetAllNightlyPackages
+    }
+    return Find-Package -Source $RELEASE_FEED_URL -AllVersion -AllowPrereleaseVersions
```

Comment:
If we aren't doing nightly/dev/beta do we still need "AllPrereleaseVersions"?

Reply:
Yes, in the case of packages that don't have a beta or GA yet, we still want to fall back to alpha.